tretools/codelists/codelist.py

```diff
@@ -206,4 +217,53 @@ def _add_X_codes_for_ICD(self, row: Dict[str, str]) -> Optional[Dict[str, str]]:
         }
 
         return new_row
+    
+
+    def _SNOMED_to_ICD10(self, SNOMED_to_ICD10_map_file: str) -> None:
+        """
+        maps SNOMED codes to their corresponding ICD10
+
+        Args:
+            SNOMED_to_ICD10_map_file (str): mapping file path
+        """
+        
+        # Using Polars for joining (otherwise longer code is needed)
+        map_file = pl.read_csv(SNOMED_to_ICD10_map_file)
+        # Assuming self.data (should be checked) is a DataFrame (or Table) containing SNOMED codes
+
+        df_data = pl.DataFrame(self.data)
 
+        df_data = df_data.with_columns([
+            pl.col('code').cast(pl.Int64) 
+        ])
+
+        # df_data = df_data.with_column(df_data['code'].cast(pl.Object))
+
+        # Joining data
+
+        df_ICD10_codes = df_data.join(map_file, left_on='code', right_on='conceptId' , how='inner') 
+        df_ICD10_codes = df_ICD10_codes.drop(['ICD10_3digit','code'])
+        df_ICD10_codes = df_ICD10_codes.rename({"mapTarget": "code"})
+        df_ICD10_codes = df_ICD10_codes.unique()
+        df_ICD10_codes = df_ICD10_codes[['code','term']]
```

Comment:
I will double check but I think it will be more memory efficient to do a swap of values via a dictionary or other data structure rather than a merge. We would be essentially loading 2 dataframes into memory which might be tricky given the size.